solr-ingest/src/test/java/edu/unc/lib/dl/data/ingest/solr/filter/SetPathFilterTest.java

```diff
@@ -15,625 +15,143 @@
-		dip.setFoxml(foxml);
+        when(pathFactory.getAncestorPids(pid)).thenReturn(pids);
+        filter.filter(dip);
+        
+        verify(idb).setAncestorPath(listCaptor.capture());
+        
+        List<String> ancestorPath = listCaptor.getValue();
+        assertTrue(ancestorPath.get(0).startsWith("1," + pids.get(0).getId()));
+        assertTrue(ancestorPath.get(1).startsWith("2," + pids.get(1).getId()));
+        assertEquals(2, ancestorPath.size());
+        verify(idb).setParentUnit(eq(pids.get(1).getId()));
+        verify(idb, never()).setParentCollection(anyString());
 
-		IndexDocumentBean idb = dip.getDocument();
-
-		SetPathFilter filter = new SetPathFilter();
-		filter.setCollectionsPid(new PID("uuid:Collections"));
-		filter.filter(dip);
-
-		assertEquals(ResourceType.Aggregate.name(), idb.getResourceType());
-		assertEquals(idb.getId(), idb.getRollup());
-		assertEquals("uuid:collection", idb.getParentCollection());
-		assertEquals(3, idb.getAncestorPath().size());
-		assertEquals("/uuid:Collections/uuid:collection/uuid:folder/uuid:aggregate", idb.getAncestorIds());
-		assertTrue(idb.getAncestorPath().contains("1,uuid:Collections"));
-		assertTrue(idb.getAncestorPath().contains("2,uuid:collection"));
-		assertTrue(idb.getAncestorPath().contains("3,uuid:folder"));
-		assertEquals(3, idb.getResourceTypeSort().intValue());
 	}
 
 	@Test
-	public void fromParentsFileTest() throws Exception {
-		DocumentIndexingPackage parentFolder = getParentFolderWithCollection();
-
-		DocumentIndexingPackage dip = factory.createDip("info:fedora/uuid:File");
-		dip.setParentDocument(parentFolder);
-		SAXBuilder builder = new SAXBuilder();
-		Document foxml = builder.build(new FileInputStream(new File("src/test/resources/foxml/imageNoMODS.xml")));
-		dip.setFoxml(foxml);
-
-		IndexDocumentBean idb = dip.getDocument();
-
-		SetPathFilter filter = new SetPathFilter();
-		filter.setCollectionsPid(new PID("uuid:Collections"));
-		filter.filter(dip);
-
-		assertEquals(ResourceType.File.name(), idb.getResourceType());
-		assertEquals(idb.getId(), idb.getRollup());
-		assertEquals("uuid:collection", idb.getParentCollection());
-		assertEquals(3, idb.getAncestorPath().size());
-		assertEquals("A1100-A800 NS final.jpg", dip.getLabel());
-		assertEquals("/uuid:Collections/uuid:collection/uuid:folder", idb.getAncestorIds());
-		assertTrue(idb.getAncestorPath().contains("1,uuid:Collections"));
-		assertTrue(idb.getAncestorPath().contains("2,uuid:collection"));
-		assertTrue(idb.getAncestorPath().contains("3,uuid:folder"));
-		assertEquals(idb.getId(), idb.getRollup());
-		assertEquals(3, idb.getResourceTypeSort().intValue());
+	public void testWorkPath() throws Exception {
+	    // Assert that the rollup is the id of the work itself
+	    List<PID> pids = makePidList(4);
+        when(pathFactory.getAncestorPids(pid)).thenReturn(pids);
+	    filter.filter(dip);
+	    
+	    verify(idb).setRollup(eq(contentObject.getPid().getId()));
```

Comment:
Verify that the ancestor path is set and that both parentCollection and parentUnit are set